### PR TITLE
[Secure Storage] Added missing README to secure storage directory.

### PR DESCRIPTION
**Pull request by JoshLind**
## Motivation

This PR adds the missing README file to the secure storage directory. This file gives a brief overview of secure storage and should help readers better understand the role secure storage plays in our ecosystem.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Yes.

## Test Plan

Viewed the file locally in markdown and everything appears to render correctly.

## Related PRs

None.

### CODE REVIEW

File: secure/storage/README.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
+---
+id: secure_storage
+title: Secure Storage
+custom_edit_url: https://github.com/libra/libra/edit/master/secure/storage/README.md
+---
+# Secure Storage
+
+Secure storage provides a secure, persistent data store for sensitive data in the Libra
+blockchain. Examples of sensitive data here include information required for safety and
```
Comment (davidiw) [inline secure/storage/README.md:9]: and run-time configuration data

File: secure/storage/README.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+- `CryptoStorage`: The CryptoStorage trait offers a cryptographic-key based storage
+abstraction for Ed25519 keys. The provided interface calls include: `create_key(..)`, to create
+a cryptographic key, `rotate_key(..)`, to rotate a cryptographic key held in storage, and `sign(..)`,
+which signs a given message using a key held in storage. The CryptoStorage API currently only
+offers support for managing two versions of a given key at any time: key versions beyond this
+cannot currently be accessed using the API.
+
+This crate provides four different secure storage implementations, each of which implements
+both `KVStorage` and `CryptoStorage`:
+- `Github`: The Github secure storage implementation provides a storage backend using a
```
Comment (davidiw) [inline secure/storage/README.md:35]: maybe too much nuance about what is required for github?

File: secure/storage/README.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+## Implementation Details
+
+This crate defines the secure storage API, made up of two separate Rust traits (interfaces):
+- `KVStorage`: The KVStorage trait offers a key-value storage abstraction. The provided
+interface calls include: `get(..)`, to retrieve the value associated with a key, `set(..)`, to set
+the value associated with a key, `available(..)`, to check if storage is currently available
+(e.g., if it has been created correctly), and `reset_and_clear(..)`, which resets and clears all
+data held in storage (this API call is only available for testing purposes.)
+- `CryptoStorage`: The CryptoStorage trait offers a cryptographic-key based storage
```
Comment (davidiw) [inline secure/storage/README.md:26]: maybe too much explicit api?

**Comment by JoshLind** [comment]
Thanks, David! I've removed the overly explicit information from the README 😄

**Pull request closed without merging**